packages/taquito-michel-codec/test/typecheck.spec.ts

```diff
@@ -1,185 +1,194 @@
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: int', () => {
+    const typedef: MichelsonType = { prim: 'int' };
+    const data: MichelsonData = { int: '0' };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: nat', () => {
+    const typedef: MichelsonType = { prim: 'nat' };
+    const data: MichelsonData = { int: '0' };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: bytes', () => {
+    const typedef: MichelsonType = { prim: 'bytes' };
+    const data: MichelsonData = { bytes: 'ABCDEF42' };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: bool', () => {
+    const typedef: MichelsonType = { prim: 'bool' };
+    const data: MichelsonData = { prim: 'True' };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: unit', () => {
+    const typedef: MichelsonType = { prim: 'unit' };
+    const data: MichelsonData = { prim: 'Unit' };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: list', () => {
+    const typedef: MichelsonType = { prim: 'list', args: [{ prim: 'string' }] };
+    const data: MichelsonData = [{ string: 'aaa' }, { string: 'bbb' }];
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: pair', () => {
+    const typedef: MichelsonType = { prim: 'pair', args: [{ prim: 'string' }, { prim: 'nat' }] };
+    const data: MichelsonData = { prim: 'Pair', args: [{ string: 'aaa' }, { int: '0' }] };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: option nat', () => {
+    const typedef: MichelsonType = { prim: 'option', args: [{ prim: 'nat' }] };
+    const data: MichelsonData = { prim: 'Some', args: [{ int: '0' }] };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: option none', () => {
+    const typedef: MichelsonType = { prim: 'option', args: [{ prim: 'nat' }] };
+    const data: MichelsonData = { prim: 'None' };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: or left', () => {
+    const typedef: MichelsonType = { prim: 'or', args: [{ prim: 'string' }, { prim: 'nat' }] };
+    const data: MichelsonData = { prim: 'Left', args: [{ string: 'aaa' }] };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: or right', () => {
+    const typedef: MichelsonType = { prim: 'or', args: [{ prim: 'string' }, { prim: 'nat' }] };
+    const data: MichelsonData = { prim: 'Right', args: [{ int: '0' }] };
+    assertDataValid(data, typedef);
+  });
+
+  it('assertDataValid: set', () => {
+    const typedef: MichelsonType = {
+      prim: 'set',
+      args: [{ prim: 'pair', args: [{ prim: 'string' }, { prim: 'nat' }] }],
+    };
+    const data: MichelsonData = [
+      { prim: 'Pair', args: [{ string: 'aaa' }, { int: '0' }] },
+      { prim: 'Pair', args: [{ string: 'bbb' }, { int: '1' }] },
+    ];
+    assertDataValid(data, typedef);
+  });
+
+  // it('assertDataValid: unsorted set', () => {
```

Comment:
Why this test and 'assertDataValid: unsorted map' are commented out?

Reply:
Well, the Tezos test set has a contract which has a clearly unsorted data but supposed to be correct at the same time. I suppose my assumption about the sorting criteria was incorrect from the beginning so I just disabled this ordering check until I figure out how to do it properly. At least I'm pretty sure that key hash comparison based on base58 form is not how it supposed to work.